tracing/tracing.go

```diff
@@ -0,0 +1,36 @@
+package tracing
+
+import (
+	propjaeger "go.opentelemetry.io/contrib/propagators/jaeger"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/trace/jaeger"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// InitTracer creates an OTel TracerProvider that exports the traces to a Jaeger collector.
+func InitTracer(serviceName, collectorEndpoint string) (tp trace.TracerProvider, closer func(), err error) {
+	disabled := collectorEndpoint == ""
+
+	tp, closer, err = jaeger.NewExportPipeline(
+		jaeger.WithCollectorEndpoint(collectorEndpoint),
+		jaeger.WithProcess(jaeger.Process{
+			ServiceName: serviceName,
+		}),
+		jaeger.WithSDK(&sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
```

Comment:
Do we really want to sample every request? Should we expose this as a flag? Or should we configure this based on the sample decision of the incoming request?

Reply:
We can use a debug flag to enable always sampler, as well as other debug related configs, i.e frame rate of profiling. But yes, we should go for a less aggressive sampling strategy.